be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -47,7 +52,64 @@ case object RollCallHandler extends MessageHandler {
   }
 
   def handleCloseRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
-    val ask: Future[GraphMessage] = dbAskWritePropagate(rpcMessage)
-    Await.result(ask, duration)
+    rpcMessage.getParamsMessage match {
+      case Some(message: Message) =>
+        val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
+        val unknownAnswer: String = "Database actor returned an unknown answer"
+
+        //Creates a channel for each attendee (of name /root/lao_id/social/PublicKeyAttendee), returns a GraphMessage
+        def createAttendeeChannels(attendees: List[PublicKey], rpcMessage: JsonRpcRequest): GraphMessage = {
+          attendees match {
+            case Nil => Left(rpcMessage)
+            case head::tail => 
+              //a closeRollCall is always sent in the lao's main channel so we are allowed to do this
+              val socialChannel: String = generateSocialChannel(rpcMessage.getParamsChannel, head)
+              val ask: Future[GraphMessage] = (dbActor ? DbActor.CreateChannel(Channel(socialChannel), ObjectType.CHIRP)).map {
+                case DbActorWriteAck() => createAttendeeChannels(tail, rpcMessage)
+                //FIXME: Once the real PoP tokens are implemented, the same person will not have access to the same channel name twice in a row, so we can remove this line
+                //however, this is to prevent errors for now within a LAO for multiple Roll Calls and the same participants since the public key will probably not change
+                case DbActorNAck(code, description) if description == "Database cannot create an already existing channel (" + socialChannel + ")" => createAttendeeChannels(tail, rpcMessage)
```

Comment:
This is dangerous. What happens if we change the message that the database returns (e.g. "can't" instead of "cannot")?

Minor: you are not using "code", you can thus change it into `case DbActorNAck(_, description) if ...`

Reply:
actually we can use the error code, as it changes between the two different outputs!